src/engine/bpmcontrol.cpp

```diff
@@ -577,13 +601,24 @@ double BpmControl::getPhaseOffset(double dThisPosition) {
     }
 
     // Get the current position of this deck.
-    double dThisPrevBeat;
-    double dThisNextBeat;
+    double dThisPrevBeat = m_pPrevBeat->get();
+    double dThisNextBeat = m_pNextBeat->get();
     double dThisBeatLength;
-    if (!getBeatContext(m_pBeats, dThisPosition,
-                        &dThisPrevBeat, &dThisNextBeat,
-                        &dThisBeatLength, NULL)) {
-        return 0;
+    if (dThisPosition > dThisNextBeat || dThisPosition < dThisPrevBeat) {
+        // There's a chance the COs might be out of date, so do a lookup.
+        qDebug() << getGroup()
```

Comment:
I guess I wasn't sure if that case should be considered a bug in my patch or not -- ie why hasn't quantizedcontrol already gotten the message about the new position?  Maybe I'll just convert it to a todo.